File: java/01-commands/tests/sourcecode/src/main/java/com/training/app/LeapYearsKata/Year.java
package com.training.app.LeapYearsKata;

public class Year {
    private int _year;

    public Year(int year) {
        _year = year;
    }

    public boolean isLeapYear() {
        if (isYearDivisibleBy(400))
            return true;
        else if (isYearDivisibleBy(4) && !isYearDivisibleBy(100))
            return true;
        return false;
    }

    private boolean isYearDivisibleBy(int toDivideBy) {
        return _year % toDivideBy == 0;
    }
}

File: nodejs/01-commands/tests/sourcecode/LeapYearKata/Year.js
class Year {
    constructor(year) {
        this._year = year;
    }

    isLeapYear() {
        const year = this._year;
        return (year % 4 === 0 && year % 100 !== 0) || (year % 400 === 0);
    }
}

module.exports = Year;